lib/Conversion/AIEVecToLLVM/AIEVecToLLVM.cpp

```diff
@@ -2412,15 +2629,12 @@ class ShuffleOpConversion
 
 void populateAIEVecToLLVMConversionPatterns(
     mlir::LLVMTypeConverter &converter, mlir::RewritePatternSet &patterns,
-    Aie2Fp32Emulation aie2Fp32EmulationOption) {
+    Aie2Fp32Emulation aie2Fp32EmulationOption, StringRef aieTarget) {
```

Comment:
We don't want to deal with this like so. We want "populate" functions per backend, so that each can be maintained in an isolated way.

You can have a look at how it's done for the [Vector to AIEVec conversion pass](https://github.com/Xilinx/mlir-aie/blob/2cb8aedb6135353227bc1bfda7f41b77c70f3895/lib/Dialect/AIEVec/Transforms/VectorToAIEVecConversions.cpp#L3098-L3186). We have `populateCommon`, `populateV1`, and `populateV2`.

Reply:
Makes sense. Codebase restructured to manage passes into pipelines: `CommonConversionPatterns`, `AIE2ConversionPatterns` and `AIE2pConversionPatterns`.